### PR TITLE
feat(): Add redux structure and home screen requests

### PR DESCRIPTION
**Pull request by AndreeaMihaelaNitu**

### CODE REVIEW

File: ios/RecipesApp/AppDelegate.mm [hunk 1]
```diff
@@ -10,7 +11,6 @@ - (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(
   // You can add your custom initial props in the dictionary below.
   // They will be passed down to the ViewController used by React Native.
   self.initialProps = @{};
-
```
Comment (timea-sillai) [inline ios/RecipesApp/AppDelegate.mm:13]: do we need this?

File: src/Pages/HomeScreen.tsx [hunk 2]
```diff
@@ -1,8 +1,49 @@
...
+  const searchMealResults = useSelector(
+    (state: any) => state.searchMealResults
+  );
+
+  React.useEffect(() => {
+    dispatch(fetchCategories());
+    dispatch(fetchRandomMeal());
+    dispatch(searchMeal("Arrabiata"));
+  }, []);
+  console.log(
```
Comment (timea-sillai) [inline src/Pages/HomeScreen.tsx]: do we need this?

File: src/Pages/HomeScreen.tsx [hunk 3]
```diff
@@ -1,8 +1,49 @@
...
+    "Categories = ",
+    categories?.categories,
+    "Loading",
+    categories?.loading,
+    " Error",
+    categories?.error
+  );
+  console.log(
+    "Random meal = ",
```
Comment (timea-sillai) [inline src/Pages/HomeScreen.tsx]: do we need this?

File: src/Pages/HomeScreen.tsx [hunk 4]
```diff
@@ -1,8 +1,49 @@
...
+    randomMeal?.randomMeal,
+    "Loading",
+    randomMeal?.loading,
+    " Error",
+    randomMeal?.error
+  );
+  console.log(
+    "Search Results = ",
+    searchMealResults?.meals,
```
Comment (timea-sillai) [inline src/Pages/HomeScreen.tsx]: do we need this?

File: src/redux/sagas/sagas.ts [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+  axios.get("https://www.themealdb.com/api/json/v1/1/search.php?s=" + mealName);
+
+// Saga worker function
+function* fetchCategoriesSaga() {
+  try {
+    yield put(setCategories(undefined, true));
+    const response: AxiosResponse<FetchCategoriesResponse> = yield call(
+      fetchCategoriesApi
+    );
+    console.log("fetchCategoriesSaga ", response.data);
```
Comment (timea-sillai) [inline src/redux/sagas/sagas.ts]: same here

File: src/redux/sagas/sagas.ts [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+  }
+}
+
+function* fetchRandomMealSaga() {
+  try {
+    yield put(setRandomMeal(undefined, true));
+    const response: AxiosResponse<FetchRandomMealResponse> = yield call(
+      fetchRandomMealApi
+    );
+    console.log("fetchRandomMealSaga ", response.data);
```
Comment (timea-sillai) [inline src/redux/sagas/sagas.ts]: same here

File: src/redux/sagas/sagas.ts [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+
+function* searchMealSaga(action: any) {
+  try {
+    yield put(setSearchMealResults(undefined, true));
+    console.log("REQUEST - ", action.payload);
+    const response: AxiosResponse<SearchMealResponse> = yield call(
+      searchMealApi,
+      action.payload
+    );
+    console.log("searchMealSaga ", response.data);
```
Comment (timea-sillai) [inline src/redux/sagas/sagas.ts]: ?

File: src/redux/sagas/sagas.ts [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+  } catch (error: any) {
+    yield put(setRandomMeal(undefined, false, error.message));
+    console.error("Error fetching meals:", error);
+  }
+}
+
+function* searchMealSaga(action: any) {
+  try {
+    yield put(setSearchMealResults(undefined, true));
+    console.log("REQUEST - ", action.payload);
```
Comment (timea-sillai) [inline src/redux/sagas/sagas.ts]: ?

File: src/redux/sagas/sagas.ts [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+    console.log("fetchRandomMealSaga ", response.data);
+    const randomMeal = response.data;
+    yield put(setRandomMeal(randomMeal, false));
+  } catch (error: any) {
+    yield put(setRandomMeal(undefined, false, error.message));
+    console.error("Error fetching meals:", error);
+  }
+}
+
+function* searchMealSaga(action: any) {
```
Comment (timea-sillai) [inline src/redux/sagas/sagas.ts]: maybe you can find a better type instead of any

File: src/Navigation/RootContainer.tsx [hunk 10]
```diff
@@ -2,11 +2,11 @@ import * as React from "react";
...
 import { createBottomTabNavigator } from "@react-navigation/bottom-tabs";
 import { BottomTabParamList } from "./NavigationTypes";
-import HomeScreen from "../Pages/HomeScreen";
-import CategoriesScreen from "../Pages/CategoriesScreen";
-import SearchScreen from "../Pages/SearchScreen";
-import UserRecipesScreen from "../Pages/UserRecipesScreen";
-import ProfileScreen from "../Pages/ProfileScreen";
+import HomeScreen from "../pages/HomeScreen";
+import CategoriesScreen from "../pages/CategoriesScreen";
+import SearchScreen from "../pages/SearchScreen";
```
Comment (timea-sillai) [inline src/Navigation/RootContainer.tsx:7]: you should stick to 1 way. you can name the folder with an uppercase letter or a lowercase letter, not both

File: src/Pages/HomeScreen.tsx (hunk 2 above)
Comment (AndreeaMihaelaNitu) [reply src/Pages/HomeScreen.tsx]: Yes, I think that logs are important for testing api requests

File: src/Navigation/RootContainer.tsx (hunk 10 above)
Comment (AndreeaMihaelaNitu) [reply src/Navigation/RootContainer.tsx:7]: All folders were converted to be lowercase written. Did I miss something?

File: src/Pages/HomeScreen.tsx (hunk 2 above)
Comment (AndreeaMihaelaNitu) [reply src/Pages/HomeScreen.tsx]: I think that we can keep all logs for this commit, as we don't have a user interface yet and I will delete all of them after the UI is ready. Do you agree?

File: ios/RecipesApp/AppDelegate.mm (hunk 1 above)
Comment (AndreeaMihaelaNitu) [reply ios/RecipesApp/AppDelegate.mm:13]: no, actually deleting that line is just a typo, I will add the space back

File: src/Navigation/RootContainer.tsx (hunk 10 above)
Comment (timea-sillai) [reply src/Navigation/RootContainer.tsx:7]: yes, 
image

**Review by timea-sillai — APPROVED** [review]

**Pull request merged**